third_party/openthread/CMakeLists.txt

```diff
@@ -26,47 +26,50 @@
 #  POSSIBILITY OF SUCH DAMAGE.
 #
 
+set(OT_15_4 ON CACHE STRING "enable 802.15.4 radio link" FORCE)
+set(OT_ANYCAST_LOCATOR ON CACHE STRING "enable anycast locator" FORCE)
+set(OT_BACKBONE_ROUTER ${OTBR_BACKBONE_ROUTER} CACHE STRING "Enable Backbone Router feature in OpenThread" FORCE)
 set(OT_BACKBONE_ROUTER_DUA_NDPROXYING ${OTBR_DUA_ROUTING} CACHE STRING "Configure DUA ND Proxy feature in OpenThread" FORCE)
 set(OT_BORDER_AGENT ON CACHE STRING "enable border agent" FORCE)
 set(OT_BORDER_ROUTER ON CACHE STRING "enable border router feature" FORCE)
 set(OT_BORDER_ROUTING ${OTBR_BORDER_ROUTING} CACHE STRING "enable border routing feature" FORCE)
 set(OT_BORDER_ROUTING_COUNTERS ${OTBR_BORDER_ROUTING_COUNTERS} CACHE STRING "enable border routing counters feature" FORCE)
 set(OT_BUILD_EXECUTABLES OFF CACHE STRING "disable building executables" FORCE)
 set(OT_BUILTIN_MBEDTLS_MANAGEMENT OFF CACHE STRING "diable mbedTLS management" FORCE)
+set(OT_CHILD_SUPERVISION ON CACHE STRING "enable child supervision" FORCE)
 set(OT_COAP ON CACHE STRING "Enable CoAP in OpenThread")
 set(OT_COAPS ON CACHE STRING "Enable secure CoAP in OpenThread")
 set(OT_COMMISSIONER ON CACHE STRING "enable commissioner")
 set(OT_DAEMON ON CACHE STRING "enable daemon mode" FORCE)
+set(OT_DATASET_UPDATER ON CACHE STRING "enable dataset updater" FORCE)
+set(OT_DNS_CLIENT ON CACHE STRING "enable DNS client" FORCE)
+set(OT_DNSSD_SERVER ${OTBR_DNSSD_DISCOVERY_PROXY} CACHE STRING "enable DNS-SD server support" FORCE)
+set(OT_ECDSA ON CACHE STRING "enable ECDSA" FORCE)
 set(OT_FIREWALL ON CACHE STRING "enable firewall feature")
+set(OT_HISTORY_TRACKER ON CACHE STRING "enable history tracker" FORCE)
 set(OT_JOINER ON CACHE STRING "enable joiner" FORCE)
 set(OT_LOG_LEVEL_DYNAMIC ON CACHE STRING "enable dynamic log level control" FORCE)
 set(OT_MAC_FILTER ON CACHE STRING "enable MAC filter" FORCE)
 set(OT_NAT64_BORDER_ROUTING ${OTBR_NAT64} CACHE STRING "enable NAT64 in border routing manager" FORCE)
 set(OT_NAT64_TRANSLATOR ${OTBR_NAT64} CACHE STRING "enable NAT64 translator" FORCE)
+set(OT_NETDATA_PUBLISHER ON CACHE STRING "enable netdata publisher" FORCE)
 set(OT_PLATFORM "posix" CACHE STRING "use posix platform" FORCE)
 set(OT_PLATFORM_NETIF ON CACHE STRING "enable platform netif" FORCE)
 set(OT_PLATFORM_UDP ON CACHE STRING "enable platform UDP" FORCE)
+set(OT_SERVICE ON CACHE STRING "enable service" FORCE)
 set(OT_SLAAC ON CACHE STRING "enable SLAAC" FORCE)
+set(OT_SRP_CLIENT ON CACHE STRING "enable SRP client" FORCE)
```

Comment:
Same as DNS client, I'm not sure if SRP client is a requirement for a border router.

Reply:
They can be useful for testing (having the CLI commands). 

I don't think there is any harm in having them enabled. 
- They add functionality/APIs (and CLI commands) but don't add any run-time overhead if their APIs are not called. 
- And we don't need to worry about code/memory size on posix.

Reply:
Just the `FORCE` may indicate the feature is required for a border router, which is not true.